src/mbgl/annotation/annotation_manager.cpp

```diff
@@ -122,8 +122,14 @@ void AnnotationManager::updateStyle(Style& style) {
         layer->sourceLayer = PointLayerID;
         layer->layout.icon.image = std::string("{sprite}");
         layer->layout.icon.allowOverlap = true;
+        layer->layerSpriteAtlas = atlas;
 
         style.addLayer(std::move(layer));
+    } else {
```

Comment:
No, that's not correct.  There is a flow whereby initially we pass in null for the sprite atlas (because we don't have any annotations), then later we set up annotations and set the sprite atlas.  You can see this flow if you open the sample app and add annotations.